File: medplum-java/medplum-java-model/src/main/java/com/fhir/MolecularSequence_ReferenceSeq.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.Integer;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableMolecularSequence_ReferenceSeq.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface MolecularSequence_ReferenceSeq {
  Optional<Molecularsequence_referenceseqStrand> strand();

  Optional<Integer> windowStart();

  Optional<List<Extension>> modifierExtension();

  Optional<CodeableConcept> chromosome();

  Optional<Integer> windowEnd();

  Optional<Reference> referenceSeqPointer();

  Optional<String> id();

  Optional<String> genomeBuild();

  Optional<Molecularsequence_referenceseqOrientation> orientation();

  Optional<String> referenceSeqString();

  Optional<List<Extension>> extension();

  Optional<CodeableConcept> referenceSeqId();

  static ImmutableMolecularSequence_ReferenceSeq.Builder builder() {
    return ImmutableMolecularSequence_ReferenceSeq.builder();
  }
}

File: medplum-typescript/src/model/fhir/MolecularSequence_ReferenceSeq.ts
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";
import { Molecularsequence_referenceseqOrientation } from "./Molecularsequence_referenceseqOrientation";
import { Molecularsequence_referenceseqStrand } from "./Molecularsequence_referenceseqStrand";
import { Reference } from "./Reference";

export interface MolecularSequence_ReferenceSeq {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    chromosome: CodeableConcept | null | undefined;
    genomeBuild: string | null | undefined;
    orientation: Molecularsequence_referenceseqOrientation | null | undefined;
    referenceSeqId: CodeableConcept | null | undefined;
    referenceSeqPointer: Reference | null | undefined;
    referenceSeqString: string | null | undefined;
    strand: Molecularsequence_referenceseqStrand | null | undefined;
    windowStart: number | null | undefined;
    windowEnd: number | null | undefined;
}
